### PR TITLE
Shapeshift tweaks.

### PR DESCRIPTION
**Pull request by AnturK**
:cl:
balance: Dying in shapeshifted form reverts you to the original one. (You're still dead)
/:cl:

I'm honestly not sure this is better.

Notransform and holder object instead of godmode. Is locking life better than locking damage ?
If you somehow manage to kill stored caster with direct damage (aka stands) you drop dead.

### CODE REVIEW

**Comment by Mercenaryblue** [comment]
Vampires could loot the armory (or the disk/upload board) then die in bat form. These items were lost forever.

This was much needed.

**Comment by Jcenderion** [comment]
No wait don't do that, or if you do make it so it wont affect the Wiz tactic of shapeshift + mindswap and that it wont make the staff of change practically worthless.

**Comment by AnturK** [comment]
Staff of change is not affected by this, just the shapeshift spell. I need to  check mindswap interaction for this though.

**Comment by AnturK** [comment]
Mindswap works fine.

**Review by ninjanomnom — APPROVED** [review]

File: code/modules/spells/spell_types/shapeshift.dm [hunk 1]
```diff
@@ -40,53 +41,122 @@
...
+	var/restoring = FALSE
+	var/datum/soullink/shapeshift/slink
+	var/obj/effect/proc_holder/spell/targeted/shapeshift/source
+
+/obj/shapeshift_holder/Initialize(mapload,obj/effect/proc_holder/spell/targeted/shapeshift/source,mob/living/caster)
+	. = ..()
+	src.source = source
+	shape = loc
+	if(!istype(shape))
+		stack_trace("shapeshift holder created outside mob/living")
```
Comment (ninjanomnom) [inline code/modules/spells/spell_types/shapeshift.dm]: This should be CRASH

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by ninjanomnom** [comment]
This honestly feels like it'd be better done with a component. This is better than current though so even as it is, is fine.

**Pull request merged**